cmd/hive/clustersync.go

```diff
@@ -0,0 +1,239 @@
+package hive
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"strconv"
+	"strings"
+	"time"
+
+	hivev1 "github.com/openshift/hive/apis/hive/v1"
+	hiveapiv1alpha1 "github.com/openshift/hive/apis/hiveinternal/v1alpha1"
+	"github.com/openshift/osdctl/pkg/printer"
+	"github.com/spf13/cobra"
+
+	// "k8s.io/apimachinery/pkg/types"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// listOptions defines the struct for running clustersync command
+type clusterSyncFailuresOptions struct {
+	includeLimitedSupport  bool
+	includeHibernating     bool
+	includeFailingSyncSets bool
+	noHeaders              bool
+	sortField              string
+	sortDescending         bool
+
+	genericclioptions.IOStreams
+	kubeCli client.Client
+}
+
+type failingClusterSync struct {
+	Name            string
+	Namespace       string
+	Timestamp       string
+	LimitedSupport  bool
+	Hibernating     bool
+	FailingSyncSets string
+}
+
+// NewCmdList implements the list command to list cluster deployment crs
+func NewCmdClusterSyncFailures(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags, client client.Client) *cobra.Command {
+	opts := newClusterSyncOptions(streams, flags, client)
+	clusterSyncCmd := &cobra.Command{
+		Use:               "clustersync-failures [flags]",
+		Short:             "list clustersync failures",
+		Args:              cobra.NoArgs,
+		Aliases:           []string{"csf"},
+		DisableAutoGenTag: true,
+		Run: func(cmd *cobra.Command, args []string) {
+			cmdutil.CheckErr(opts.complete(cmd, args))
+			cmdutil.CheckErr(opts.run())
+		},
+	}
+	clusterSyncCmd.Flags().BoolVarP(&opts.includeLimitedSupport, "include-limited-support", "L", false, "Include clusters in limited support.")
+	clusterSyncCmd.Flags().BoolVarP(&opts.includeHibernating, "include-hibernating", "H", false, "Include hibernating clusters.")
+	clusterSyncCmd.Flags().BoolVarP(&opts.includeFailingSyncSets, "include-failing-syncsets", "F", false, "Include failing syncsets.")
+	clusterSyncCmd.Flags().BoolVar(&opts.noHeaders, "no-headers", false, "Don't print headers.")
+	clusterSyncCmd.Flags().BoolVar(&opts.sortDescending, "sort-desc", false, "Sort in a descending order. Default false.")
+	clusterSyncCmd.Flags().StringVar(&opts.sortField, "sort-by", "timestamp", "Sort by field.")
+
+	return clusterSyncCmd
+}
+
+func newClusterSyncOptions(streams genericclioptions.IOStreams, flags *genericclioptions.ConfigFlags, client client.Client) *clusterSyncFailuresOptions {
```

Comment:
Do we even need `flags`/can we remove it?

Reply:
I think we could remove `newClusterSyncOptions` as a whole, and just do, I don't see the point in a function that just returns the values it gets, and this boilerplate doesn't seem to be necessary due to interfacing or anything else at first glance.

```
opts := &clusterSyncFailuresOptions{
		IOStreams: streams,
		kubeCli:   client,
	}
```

Reply:
Thank you. I removed the function.